lgc/builder/llpcYCbCrAddressHandler.cpp

```diff
@@ -0,0 +1,196 @@
+*/
+#include "llpcYCbCrAddressHandler.h"
+#include "llpcGfxRegHandler.h"
+#include "llpcInternal.h"
+
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
+
+using namespace lgc;
+using namespace llvm;
+
+// =====================================================================================================================
+// The class is used for calculating and mantain the base address of each plane in YCbCr image.
+void YCbCrAddressHandler::GenBaseAddress(
+    uint32_t planeNum) // The number of planes
+{
+    // For YCbCr, the possible plane numbers are among 1 and 3.
+    assert(planeNum > 0 && planeNum < 4);
+
+    if (planeNum > 0)
+    {
+        // m_PlaneBaseAddresses[0] is the same as original base addressed as passed in SRD
+        m_PlaneBaseAddresses.push_back(m_pRegHelper->GetReg(SqRsrcRegs::BaseAddress));
+        if (planeNum > 1)
+        {
+            // m_PlaneBaseAddresses[1] = m_PlaneBaseAddresses[0] + addr_256B(m_pPitchY * m_pHeightY)
+            m_PlaneBaseAddresses.push_back(m_pBuilder->CreateAdd(m_PlaneBaseAddresses[0],
+                                           m_pBuilder->CreateLShr(m_pBuilder->CreateMul(m_pPitchY, m_pHeightY),
+                                           m_pBuilder->getInt32(8))));
+            if (planeNum > 2)
+            {
+                // m_PlaneBaseAddresses[2] = m_PlaneBaseAddresses[1] + addr_256B(m_pPitchCb * m_pHeightCb)
+                m_PlaneBaseAddresses.push_back(m_pBuilder->CreateAdd(m_PlaneBaseAddresses[1],
+                                               m_pBuilder->CreateLShr(m_pBuilder->CreateMul(m_pPitchCb, m_pHeightCb),
+                                               m_pBuilder->getInt32(8))));
+            }
+        }
+    }
+}
+
+// =====================================================================================================================
+// Power2align operation
+Value* YCbCrAddressHandler::Power2Align(
+    Value*   pX,    // [in] Value needs to be aligned
+    uint32_t align) // Align base
+{
+    // Check if align is power2 aligned
+    assert(align != 0 && (align & (align-1)) == 0);
+
+    uint32_t alignM1 = align - 1;
+    uint32_t alignM1Inv = ~alignM1;
+    Value* pResult = m_pBuilder->CreateAdd(pX, m_pBuilder->getInt32(alignM1));
+    return m_pBuilder->CreateAnd(pResult, m_pBuilder->getInt32(alignM1Inv));
+}
+
+// =====================================================================================================================
+// Calculate height and pitch
+// Note: Need pRegHelper to be bounded
+void YCbCrAddressHandler::GenHeightAndPitch(
+    uint32_t bits,          // Channel bits
+    uint32_t bpp,           // Bits per pixel
+    uint32_t xBitCount,     // Effective channel bits
+    bool     isTileOptimal, // Is tiling optimal
+    uint32_t planeNum)      // Number of planes
+{
+    switch (m_pGfxIpVersion->major)
+    {
+    case 9:
+    {
+        // pHeight = SqRsrcRegs::Height
```

Comment:
pHeight -> height
In comments, don't use "p" for variables because this is a just description. Apply this to following codes.